src/app/devices/deviceContent/components/digitalTwin/digitalTwinInterfaces.tsx

```diff
@@ -110,35 +208,38 @@ export const DigitalTwinInterfaces: React.FC<DigitalTwinInterfacesProps> = props
                 tooltip={ResourceKeys.settings.questions.questions.documentation.text}
             />
-
-            <section className="device-detail">
-                {props.dcm && <Label className="dcm-info">{t(ResourceKeys.digitalTwin.dcm, {modelId: props.dcm })}</Label>}
-                {modelContents.length !== 0 &&
-                    <div className="list-detail">
-                        <DetailsList
-                            onRenderItemColumn={renderItemColumn()}
-                            items={modelContents}
-                            columns={getColumns()}
-                            selectionMode={SelectionMode.none}
-                        />
-                    </div>
-                }
-            </section>
+            {props.isDigitalTwinLoading ?
+                <MultiLineShimmer/> :
+                <section className="device-detail">
+                    <h4>{t(ResourceKeys.digitalTwin.steps.first)}</h4>
+                    <MaskedCopyableTextFieldContainer
+                        ariaLabel={t(ResourceKeys.digitalTwin.modelId)}
+                        label={t(ResourceKeys.digitalTwin.modelId)}
+                        value={props.modelId}
+                        allowMask={false}
+                        readOnly={true}
+                    />
+                    {renderModelDefinition()}
+            </section>}
         </>
     );
 };
 
 export type DigitalTwinInterfacesContainerProps = RouteComponentProps;
 export const DigitalTwinInterfacesContainer: React.FC<DigitalTwinInterfacesContainerProps> = props => {
-    const digitalTwinInterfacesWrapper = useSelector(getDigitalTwinInterfacePropertiesWrapperSelector);
+    const digitalTwinSynchronizationStatus = useSelector(getDigitalTwinSynchronizationStatusSelector);
+    const modelDefinitionSynchronizationStatus = useSelector(getModelDefinitionSyncStatusSelector);
     const dispatch = useDispatch();
 
     const viewProps = {
-        dcm: useSelector(getDigitalTwinDcmNameSelector),
-        isLoading: digitalTwinInterfacesWrapper &&
-            digitalTwinInterfacesWrapper.synchronizationStatus === SynchronizationStatus.working,
-        nameToIds: useSelector(getDigitalTwinComponentNameAndIdsSelector),
-        refresh: (deviceId: string) => dispatch(getDigitalTwinInterfacePropertiesAction.started(deviceId)),
+        componentNameToIds: useSelector(getComponentNameAndInterfaceIdArraySelector),
+        isDigitalTwinLoading: digitalTwinSynchronizationStatus === SynchronizationStatus.working,
+        isModelDefinitionLoading: modelDefinitionSynchronizationStatus === SynchronizationStatus.working,
+        modelDefinitionWithSource: useSelector(getModelDefinitionWithSourceSelector),
+        modelId: useSelector(getDigitalTwinModelId),
+        retrieveComponents: (deviceId: string, interfaceId: string) => dispatch(getModelDefinitionAction.started({digitalTwinId: deviceId, interfaceId})),
```

Comment:
why do we need to pass this dispatch props down to the DigitalTwinInterfaces FC ? you can call this actions from child Fc through useDispatch directly